### PR TITLE
Notify operator inactivity refund

### PR DESCRIPTION
**Pull request by dimpar**
Refs: https://github.com/keep-network/keep-core/issues/2839

Refunding ETH for calling `notifyOperatorInactivity`. Gas offset was set to `85000` with a try&error approach. The notifier will be a bit net-positive, but less than `0.002ETH` when the gas price is 200 gwei. In reality, it will be even lower because tests use some slack.
Gas offset is a gov param and can be adjusted according to the current market conditions.

### CODE REVIEW

**Comment by pdyraga** [comment]
👀

File: solidity/ecdsa/contracts/WalletRegistry.sol [hunk 1]
```diff
@@ -76,10 +76,16 @@ contract WalletRegistry is
     ///         other member that will call the DKG approve function.
     uint256 public dkgResultSubmissionGas = 275000;
 
-    // @notice Gas meant to balance the DKG result approval's overall cost. Can
-    //         be updated by the governace based on the current market conditions.
+    /// @notice Gas is meant to balance the DKG result approval's overall cost.
```
Comment (pdyraga) [inline solidity/ecdsa/contracts/WalletRegistry.sol:79]: Should it be "Gas is meant to..." or "Gas meant to"? My understanding is that the second is correct, given that we are explaining what `dkgResultApprovalGasOffset` is for.

```suggestion
    /// @notice Gas meant to balance the DKG result approval's overall cost.
```

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol [hunk 2]
```diff
@@ -71,6 +71,9 @@ contract WalletRegistryGovernance is Ownable {
     address payable public newReimbursementPool;
     uint256 public reimbursementPoolChangeInitiated;
 
+    uint256 public notifyOperatorInactivityGasOffsetChangeInitiated;
+    uint256 public newNotifyOperatorInactivityGasOffset;
```
Comment (pdyraga) [inline solidity/ecdsa/contracts/WalletRegistryGovernance.sol:75]: Can we place them above `newReimbursementPool`, right after DKG result approval gas offset fields?

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol [hunk 3]
```diff
@@ -172,6 +175,14 @@ contract WalletRegistryGovernance is Ownable {
     );
     event ReimbursementPoolUpdated(address reimbursementPool);
 
+    event NotifyOperatorInactivityGasOffsetUpdateStarted(
+        uint256 notifyOperatorInactivityGasOffset,
+        uint256 timestamp
+    );
+    event NotifyOperatorInactivityGasOffsetUpdated(
+        uint256 notifyOperatorInactivityGasOffset
+    );
```
Comment (pdyraga) [inline solidity/ecdsa/contracts/WalletRegistryGovernance.sol:184]: Can we place them above `ReimbursementPoolUpdateStarted`, right after events for DKG approval gas offset?

File: solidity/ecdsa/deploy/02_deploy_reimbursement_pool.ts [hunk 4]
```diff
@@ -14,6 +18,11 @@ const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
     log: true,
   })
 
+  await deployerSigner.sendTransaction({
+    to: ReimbursementPool.address,
+    value: ethers.utils.parseEther("100.0"), // Send 100.0 ETH
+  })
```
Comment (pdyraga) [inline solidity/ecdsa/deploy/02_deploy_reimbursement_pool.ts:24]: Can we do it only if `deployments.getNetworkName() === "hardhat"`?

For mainnet deployment, we definitely not want to send 100 ETH from the deployer account.

File: solidity/ecdsa/test/WalletRegistryGovernance.test.ts [hunk 5]
```diff
@@ -2418,4 +2419,143 @@ describe("WalletRegistryGovernance", async () => {
...
+      it("should revert", async () => {
+        await expect(
+          walletRegistryGovernance
+            .connect(thirdParty)
+            .beginNotifyOperatorInactivityGasOffsetUpdate(100)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner and value is correct", () => {
```
Comment (pdyraga) [inline solidity/ecdsa/test/WalletRegistryGovernance.test.ts:2434]:
```suggestion
    context("when the caller is the owner", () => {
```

We do not really validate the value.

**Review by pdyraga — APPROVED** [review]
Some non-blocking comments. @dimpar please let me know what you think and I can address them in my PR - I need to add `withdrawIneligibleRewards` to `WalletRegistry`.

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol (hunk 2 above)
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistryGovernance.sol:75]: Sure

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol (hunk 3 above)
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistryGovernance.sol:184]: Sure

File: solidity/ecdsa/deploy/02_deploy_reimbursement_pool.ts (hunk 4 above)
Comment (dimpar) [reply solidity/ecdsa/deploy/02_deploy_reimbursement_pool.ts:24]: Yea, good catch, certainly it was added only for the testing hardhat purposes.

File: solidity/ecdsa/test/WalletRegistryGovernance.test.ts (hunk 5 above)
Comment (dimpar) [reply solidity/ecdsa/test/WalletRegistryGovernance.test.ts:2434]: Right, copy pasta from other tests.

**Comment by dimpar** [comment]
> 

Sure, I've replied to your comments. I can create a new PR addressing them or also totally fine if you address them in your PR. Thanks

**Pull request merged**